File: backend/app/models/inquiry.py
from datetime import datetime

from sqlalchemy import DateTime, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class Inquiry(Base):
    __tablename__ = "inquiries"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    submission_number: Mapped[str] = mapped_column(String(40), unique=True, index=True, nullable=False)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    company: Mapped[str] = mapped_column(String(150), nullable=False)
    email: Mapped[str | None] = mapped_column(String(180), nullable=True)
    phone: Mapped[str | None] = mapped_column(String(80), nullable=True)
    attachment_url: Mapped[str | None] = mapped_column(String(500), nullable=True)
    attachment_name: Mapped[str | None] = mapped_column(String(255), nullable=True)
    attachment_content_type: Mapped[str | None] = mapped_column(String(120), nullable=True)
    product_slug: Mapped[str | None] = mapped_column(String(220), nullable=True)
    product_code: Mapped[str | None] = mapped_column(String(24), nullable=True)
    solution_slug: Mapped[str | None] = mapped_column(String(220), nullable=True)
    message: Mapped[str] = mapped_column(Text, nullable=False)
    source_page: Mapped[str] = mapped_column(String(500), nullable=False)
    status: Mapped[str] = mapped_column(String(40), default="new", nullable=False)
    crm_status: Mapped[str] = mapped_column(String(40), default="pending", nullable=False)
    crm_attempts: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    crm_last_error: Mapped[str | None] = mapped_column(Text, nullable=True)
    crm_synced_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, nullable=False)

File: management/src/types/inquiry.ts
export type InquiryStatus = 'new' | 'contacted' | 'quoted' | 'won' | 'lost' | 'archived'

export interface InquiryNote {
  id: number
  inquiry_id: number
  admin_user_id: number
  note: string
  created_at: string
  admin_username: string
}

export interface ManagementInquiry {
  id: number
  submission_number: string
  name: string
  company: string
  email: string | null
  phone: string | null
  attachment_url: string | null
  attachment_name: string | null
  attachment_content_type: string | null
  product_slug: string | null
  product_code: string | null
  solution_slug: string | null
  message: string
  source_page: string
  status: InquiryStatus
  crm_status: 'pending' | 'retrying' | 'synced' | 'failed' | 'not_configured'
  crm_attempts: number
  crm_last_error: string | null
  crm_synced_at: string | null
  created_at: string
  notes: InquiryNote[]
}
